core/embed/prodtest/optiga_prodtest.c

```diff
@@ -143,6 +143,12 @@ void pair_optiga(void) {
   uint8_t secret[SECRET_OPTIGA_KEY_LEN] = {0};
 
   if (secret_optiga_get(secret) != sectrue) {
+    if (secret_optiga_writable() != sectrue) {
```

Comment:
due to this, the bootloader can be re-locked by prodtest on the F4 but not on the U5, correct?

shouldn't we add re-locking capability to prodtest for U5 too? we'd need a command to explicitly erase the secret area
(or maybe we just stuff `secret_erase` here?)

Reply:
Correct.

I would say that re-locking bootloader is in fact unwanted. On F4 we cant really prevent it though, custom firmware can erase the sector whenever it likes and write whatever it wants there. On U5 the secret area contains other stuff we'd need to erase, namely bootloader and firmware monotonic counters, which shouldnt ideally be possible. We could attempt to re-populate the counters but that still leaves space for attacks via power disconnection in right time.

Reply:
i mean, if someone gets ahold of a leaked prodtest, they can use it to erase the secret sector, which will also lose them the pairing secret, so their device no longer registers as authentic
OTOH the factory might want the ability to re-pair a specific U5 to a different Optiga
(probably no need to solve that right now though -- we can produce an updated prodtest that does this at some later point)

Reply:
Yeah i would wait with adding this until there is a real need from factory.